test/test_mahalanobis_mixin.py

```diff
@@ -281,5 +281,15 @@ def test_transformer_is_2D(estimator, build_dataset):
 
   # test that it works for 1 feature
   trunc_data = input_data[..., :1]
+  # we drop duplicates that might have been formed, i.e. of the form
+  # aabc or abcc or aabb for quadruplets, and aa for pairs.
+  slices = {4: [slice(0, 2), slice(2, 4)], 2: [slice(0, 2)]}
+  if trunc_data.ndim == 3:
+    for slice_idx in slices[trunc_data.shape[1]]:
+      pairs = trunc_data[:, slice_idx, :]
+      diffs = pairs[:, 1, :] - pairs[:, 0, :]
+      to_keep = np.nonzero(diffs.ravel())
```

Comment:
This is a bit difficult to parse. Why do we need these slices? I am a bit lazy to check but maybe it should be made more clear even if it is less efficient (this is a small 1D dataset anyway so we don't care)

also maybe removing things that are very close to being the same (as opposed to exactly the same) would be more robust

Reply:
I agree it's difficult to parse, I'll change the test copying/pasting for the quadruplets/pairs case.

Reply:
Done, in commit https://github.com/metric-learn/metric-learn/pull/162/commits/31072d326620dce295669a45108f336d138459a0
What do you think ?